resources/newssite/news-nuxt/layouts/default.vue

```diff
@@ -0,0 +1,52 @@
+<script setup>
+import { ref, inject, onMounted } from "vue";
+import { useRoute } from "nuxt/app";
+import { nextTick, watch } from "#imports";
+import styles from "news-site-css/dist/layout.module.css";
+
+const showMessage = ref(false);
+const route = useRoute();
+
+const { content, links } = inject("data");
+
+onMounted(() => {
+    showMessage.value = content[route.name].message;
+});
+
+const closeMessage = () => {
+    showMessage.value = false;
+};
+
+watch(
+    route,
+    // eslint-disable-next-line @typescript-eslint/no-unused-vars
+    (value) => {
+        if (document.getElementById("page")) {
+            if (!route.hash) {
+                document.getElementById("page").scrollTo(0, 0);
+            } else {
+                const elementId = route.hash.split("#")[1];
+                nextTick(() => {
+                    document.getElementById(elementId).scrollIntoView();
+                });
+            }
+        }
+    },
+    { deep: true, immediate: true }
+);
```

Comment:
I know it's not a change in this PR, but this could likely be refactored into `scrollBehavior` in `app/router.options.ts`

Reply:
yeah - for now I also made it a composable... maybe something in between? 
happy to make that simpler